### PR TITLE
wip Nfultz/next

### PR DESCRIPTION
**Pull request by nfultz**

### CODE REVIEW

**Comment by nfultz** [comment]
Just a heads up, `pmatch` is a base R function. Its fine to collide with it, but if there's an alternative name for this distribution you were considering, it's one more point for it.

**Comment by ben-oneill** [comment]
Thanks for spotting that.  In that case, perhaps change it to matching (i.e., dmatching, pmatching, qmatching, rmatching, etc.).


________________________________

File: R/HDR.R [hunk 1]
```diff
@@ -579,3 +580,152 @@ print.hdr <- function(x, ...) {
...
+      MEAN <- 1 + n*prob - prob^n
+      VAR  <- 1 - prob^(2*n) + n*(prob - prob^2 - prob^(n-1) - prob^n + 2*prob^(n+1))
+      LOGPROBS.TOTAL <- dnorm(0:(n*m), mean = m*MEAN, sd = sqrt(m*VAR), log = TRUE)
+      LOGPROBS.TOTAL <- LOGPROBS.TOTAL - matrixStats::logSumExp(LOGPROBS.TOTAL)
+      
+    } else {
+      
+      #Deal with case where n = Inf and prob = 0
+      if ((n == Inf)&(prob == 0)) {
+        LOGPROBS <- dpois(x, lambda = 1, log = TRUE) }
```
Comment (nfultz) [inline R/HDR.R]: @ben-oneill 
R CMD Check says that `x` is not in scope here. This was probably copy-pasted from somewhere else, and you missed an edit. Same issue in `qmatching()`

For HDR, if its asymptotically poission, could just return the result of HDR.poission. Same strategy for qmatching.

**Comment by nfultz** [comment]
I pulled in your other changes and fixed the warnings; haven't checked the outputs specifically though. I sent a build to rhub and winbuilder, assumming those are ok you can submit to cran.

**Pull request merged**